testproject/Assets/Tests/Runtime/MultiprocessRuntime/Helpers/BuildMultiprocessTestPlayer.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.IO;
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEditor.Build.Reporting;
+#endif
+using UnityEngine;
+
+/// <summary>
+/// This is needed as Unity throws "An abnormal situation has occurred: the PlayerLoop internal function has been called recursively. Please contact Customer Support with a sample project so that we can reproduce the problem and troubleshoot it."
+/// when trying to build from Setup() steps in tests.
```

Comment:
did you reach anybody about this? do we know what's going on here? 👀

Reply:
yep, it's kind of expected when you think about it. One wouldn't be expected to build things from players...
One possible solution would have been to have editor tests that call "EnterPlayMode" (https://docs.unity3d.com/2018.3/Documentation/ScriptReference/TestTools.EnterPlayMode.html)
But that would have prevented us from running tests in remote builds bound us to an editor.